### ISSUE TITLE
Select Fieldtype - Can't render and issues when adding new values

### ISSUE THREAD

**Issue by Pixney-William** [statamic/cms#1504]
**My select field type is names "categories".**

**If i add a new value i get issues. See below, if i enter cat and press enter**:
![image](https://user-images.githubusercontent.com/31292231/76679495-5d903400-65e1-11ea-95d4-b1527e6a9a38.png)

If you then click back and add dog and hit enter this shows up:
![image](https://user-images.githubusercontent.com/31292231/76679548-c1b2f800-65e1-11ea-9a14-330bce54f9ce.png)

You are now not able to click(remove) the first tag that showed up with just the close icon but no value/content.

**If you save, you get a null value saved as below, the select field type is now rendered at all in my blueprint:**
![image](https://user-images.githubusercontent.com/31292231/76679469-0be7a980-65e1-11ea-9c0a-99cd440bbefa.png)
![image](https://user-images.githubusercontent.com/31292231/76679470-1144f400-65e1-11ea-93eb-e6f100553bd6.png)
![image](https://user-images.githubusercontent.com/31292231/76679477-2cafff00-65e1-11ea-8eb5-02a4f8306608.png)

# If i don't try to enter new values, but just use those already in the select box ie:
![image](https://user-images.githubusercontent.com/31292231/76679617-4d2c8900-65e2-11ea-852f-073e1c76dd8c.png)

It all works fine, but when i am trying to use it in my view i get an error:

```
{{ categories }}
    {{ foreach:options }}
    {{ key }} {{ value }}
    {{ /foreach:options }}
{{ /categories }}
```
or 
`{{ categories }}`

I get : 
![image](https://user-images.githubusercontent.com/31292231/76679646-8b29ad00-65e2-11ea-964b-cf5078ffa0eb.png)

**Comment by goellner** [comment]
Same problem here, took a while to narrow it down to the "select" fieldtype.

**Comment by goellner** [comment]
One thing I found, while using it headless: 

When I access the /globals api endpoint, the same error gets thrown, even though there is no `select` field in my globals. When I remove the field from the .yaml file, then the error goes away, but the data is missing in the endpoint.

I have no issues saving/editing the entry, all the data in the .md file looks correct. 

Does Statamic parse all fieldsets of the whole site when accessing the API?

**Comment by jackmcdade** [comment]
Fixed in 06ddc246f38a664b8d05ec9344f9071866f50c1c
